backend/data_handling/data_processing.py

```diff
@@ -0,0 +1,79 @@
+import pandas as pd
+from data_handling.data_maps import detailed_category_map, sum_category_map
+import random
+
+
+"""
+helper function to filter and simplify the JSON returned by the 
+[transactions/sync] endpoint into the relevant fields of
+amount, date, currency, primary, and detailed categories. 
+"""
+def clean_transaction_data(transaction_json):
+    cleaned_data = []
+    for transaction in transaction_json['transactions']:
+        # filter out transactions where money is deposited to account since we're concerned with expenditures
+        if transaction["amount"] > 0 :
+            clean_transaction = {
+                # for the sake of the demo, amount spent on transactions is random to provide differeneces to analyze
+                "amount": round(random.uniform(0.01, 5000), 2),
+                "date": transaction["date"],
+                "currency": transaction["iso_currency_code"],
+                "detailed": transaction["personal_finance_category"]["detailed"],
```

Comment:
Especially things like this `transaction['personal_finance_category']['detailed']` would crash I think if `personal_finance_category` doesn't exist. Do we need to be more defensive here?